src/client/Cosmetic.ts

```diff
@@ -0,0 +1,99 @@
+import { base64url } from "jose";
+import { z } from "zod";
+import rawTerritoryPatterns from "../../resources/cosmetic/cosmetic.json";
+
+export const CosmeticsSchema = z.object({
+  role_group: z.record(z.string(), z.array(z.string())).optional(),
```

Comment:
I think we should be consistent about using postfix notation for arrays:
```suggestion
  role_group: z.record(z.string(), z.string().array()).optional(),
```